hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -4944,6 +4944,24 @@ public CompletableFuture<FSDataInputStream> build() throws IOException {
 
   }
 
+  /**
+   * Return path of the enclosing root for a given path
+   * The enclosing root path is a common ancestor that should be used for temp and staging dirs
+   * as well as within encryption zones and other restricted directories.
+   *
```

Comment:
nit: add a <p> for line breaks in javadocs and IDE rendering

Reply:
not sure what the ask is here, maybe something was missed

Reply:
ahhh based on next comment going to assume its "."

Reply:
if my assumption is correct then done, if not reopen and lmk